### ISSUE TITLE
[V2] Improve prediction

### ISSUE THREAD

**Issue by gavinhenderson** [AceCentre/Echo#13]
Currently the prediction algorithm is something I just made up. It is 'fast' but it is a very naive implementation of prediction.

Before any work is done on improving prediction i think we need to create a basic test harness that can evaluate any given prediction otherwise we will just be shooting in the dark. The kind of thing I have in mind is we run a few sentences through the prediction engine and count how many clicks it would take to get to the target sentence. Its not a perfect metric but it will at least tell us if we are going in the right direction.

For ref will wrote up these notes: #2 

@willwade Open question, whats your thoughts regarding if this should block release or not? Do we push this to a future version or do we improve the prediction ahead of release?

**Comment by gavinhenderson** [comment]
Closing in favour of #33 #34 #40
